src/node.cpp

```diff
@@ -394,7 +395,8 @@ Node* Node::unserialize(MegaClient* client, string* d, node_vector* dp)
             return NULL;
         }
 
-        handle ph = MemAccess::get<handle>(ptr);
+        handle ph = 0;
+        memcpy((char*)&ph, ptr, MegaClient::NODEHANDLE);
```

Comment:
Good catch, indeed!! I wonder how it happened that we never got any report about this issue, since the `ets` the handle would not match the actual public handle. Probably it's because none uses the PRO feature to set the expire timestamps for public links :( (so it's 0 and the handle gets its higher bytes set to zero by coincidence).